appveyor.yml

```diff
@@ -5,19 +5,94 @@ environment:
   # MinGW
   - name: win32
     platform: mingw
-    qt: 5.9\mingw53_32
+    qt: 5.5\mingw492_32
+    tools: mingw492_32
+  # MinGW
+  - name: win32
+    platform: mingw
+    qt: 5.6\mingw49_32
+    tools: mingw492_32
+  # MinGW
+  #  - name: win32
+  #    platform: mingw
+  #    qt: 5.9\mingw53_32
+  #    tools: mingw530_32
+  # MSVC x32
+  - name: win32
+    platform: x86
+    qt: 5.6\msvc2015
   # MSVC x64
   - name: win64
     platform: amd64
-    qt: 5.9\msvc2015_64
+    qt: 5.6\msvc2015_64
 
 init:
   - if %platform%==mingw set PATH=%PATH:C:\Program Files\Git\usr\bin;=%
-  - if %platform%==mingw set PATH=C:\Qt\Tools\mingw530_32\bin;%PATH%
+  - if %platform%==mingw set PATH=C:\Qt\Tools\%tools%\bin;%PATH%
   - set PATH=C:\Qt\%qt%\bin;%PATH%
   - if not %platform%==mingw call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" %platform%
 
 # Flesh out with amd64/msvc cases later.
 build_script:
-  - qmake GPSBabel.pro
-  - if not %platform%==mingw (nmake) else (mingw32-make)
+  - ps: |
+      if (($env:platform -eq "mingw") -and ($env:qt -eq "5.5\mingw492_32"))
+      {
+        Get-Location
```

Comment:
Should the bulk of this perhaps be in a tools/something.bat file with the release Qt directory passed in so we can avoid tying that into Appveyor?  ("No" is an OK answer.)

Reply:
maybe.  I think the most important reason to do that would be to allow the script to run outside of appveyor.  One can also argue against having lengthy scripts in CI configuration files.  Moving it has the potential to affect appveyor error handling.  I'm not there yet.

I did modify appveyor.yml to set ps variables for the shadow directories, so at least they only have to be changed in one place.

Note that we don't have to use the default Qt creator shadow directory names.  We could use Qt version and compiler independent names in appveyor.yml and setup.iss.in.

Reply:
That's just historical. The batch files have tended to be "automate what I just did" so the last few releases have been "run Qt Creator and accept the defaults and then feed that to innosetup and then give more steroids to the pre-Qt batch file" It was thus approximately the "if mingw" case below.